### PR TITLE
Created sources/sun2020

### PR DESCRIPTION
**Pull request by RyanBerger98**
Pull Request to include MicroHaps included in Sun2020

### CODE REVIEW

**Comment by RyanBerger98** [comment]
This is ready for review!

**Review by standage — COMMENTED** [review]
Looks good! After looking at the new `marker.tsv` file, I realized that none of the new markers have A<sub>e</sub> values or I<sub>n</sub> values. This reminded me of a circular dependency in the build process that has required me to do a kind of two-step rebuild.

- The first step is the documented process you just completed: run the Snakemake build workflow in the dbbuild dir, and copy the updated .tsv files to the microhapdb/data/ directory.
- The second is to go to the The 1000 Genomes project source dir (1kgp), run the `get-marker-rsids.py` script to create a new `marker-rsids-MicroHapDB-*.tsv` file that includes the recently added markers, and re-run **that** directory's build process, and then re-run the main dbbuild process again.

It's a little kludgy, and I didn't realize until just now that it's not well documented. But if you could indulge me in going through this second step to compute the various statistics, that would be great. In the mean time, I'll see what I can do to improve the process and its documentation.

**Comment by standage** [comment]
We'll also need to make some minor adjustments to the test suite to accommodate the new markers, but let's fix all the build issues first.

**Comment by RyanBerger98** [comment]
Followed the instruction to rebuild the 1kgp directory and then rebuilt the MicroHapDB. Moved the appropriate `.tsv` files to the `microHapDB/data` directory. It seems that `marker.tsv` now contains the A<sub>e</sub> and I<sub>n</sub> values it was previously missing.

**Comment by standage** [comment]
Ok, I've confirmed the build on my end! Let me provide some pointers for updating the test suite. (In the mean time, I'll note that it looks like the `sources/1kgp/marker-aes.tsv` file was not copied over along with the other files.)

**Comment by standage** [comment]
Some test failures (like `microhapdb/tests/test_marker.py::test_all_sources`) seem to be the result of inconsequential rounding issues. We can update these accordingly to the new expected value.

**Comment by standage** [comment]
The test suite also runs "doctests", which evaluate the code in the function docstrings and tests whether the expected output matches the observed output. Some of these failures are probably due to the index number for each row in the dataframe being shifted due to the addition of new markers. I typically update just the index numbers to make sure the rest of the string is unchanged.

**Comment by standage** [comment]
Take a look at the failing tests and see if you can fix the ones that fall into these categories. If there are any remaining after that, I'm happy to chat if the cause of the failure is unclear.

File: microhapdb/tests/test_cli.py [hunk 1]
```diff
@@ -109,7 +109,7 @@ def test_main_marker_noargs(capsys):
     microhapdb.cli.main(args)
     out, err = capsys.readouterr()
     outlines = out.strip().split('\n')
-    assert len(outlines) == 1 + 198 + 15 + 40 + 26 + (11 - 1) + 10 + 118 + 90
+    assert len(outlines) == 1 + 198 + 15 + 40 + 26 + (11 - 1) + 10 + 118 + 90 + 25
```
Comment (RyanBerger98) [inline microhapdb/tests/test_cli.py:112]: There were 25 new haplotypes added, so we add a `+25` here

File: microhapdb/tests/test_cli.py [hunk 2]
```diff
@@ -129,6 +129,7 @@ def test_main_marker_query(capsys):
   mh19KK-056 MHDBM-d6ff8635    GRCh38 chr19                                                4852124,4852324 2.4391 0.0773  0.1760                         ALFRED
   mh19CP-007 MHDBM-49dbcc57    GRCh38 chr19                                     14310739,14310772,14310780 3.0813 0.0466  0.0776                         ALFRED
 mh19USC-19pB MHDBM-76427274    GRCh38 chr19                   16040864,16040894,16040899,16040921,16040929 3.5107 0.1647  0.0731   10.1016/j.fsigen.2019.102213
+ mh19ZHA-006 MHDBM-5f597339    GRCh38 chr19          20579862,20579863,20579879,20579892,20579916,20579924 3.8739 0.1681 -0.0032   10.1016/j.fsigen.2020.102255
```
Comment (RyanBerger98) [inline microhapdb/tests/test_cli.py:132]: `mh19ZHA-006` is a new marker that we added, and it resides on the 19th chromosome. Updated to be included in the test.

File: microhapdb/tests/test_cli.py [hunk 3]
```diff
@@ -288,7 +289,7 @@ def test_main_marker_view_bad():
     ('--population=SA000009J', '--marker=mh13KK-218', None, 15),
     (None, '--marker=mh13KK-218', '--allele=C,T,C,T', 97),
     (None, '--marker=mh14PK-72639', None, 217),
-    (None, None, None, 126289)
+    (None, None, None, 130420)
```
Comment (RyanBerger98) [inline microhapdb/tests/test_cli.py:292]: Updated total length

File: microhapdb/tests/test_cli.py [hunk 4]
```diff
@@ -336,6 +337,7 @@ def test_ae_pop(capsys):
 mh18PK-87558 MHDBM-1e5374f1    GRCh38 chr18 1960542,1960557,1960561,1960566,1960582,1960588 1.9767 0.1325 0.0494 10.1016/j.fsigen.2018.05.008
 mh18USC-18pA MHDBM-56dfa93b    GRCh38 chr18         5280017,5280020,5280070,5280071,5280093 1.8402 0.2130 0.1818 10.1016/j.fsigen.2019.102213
   mh18CP-005 MHDBM-a85754d3    GRCh38 chr18                 8892864,8892893,8892896,8892907 3.3873 0.0904 0.0059                       ALFRED
+ mh18ZHA-004 MHDBM-db649dba    GRCh38 chr18             14315931,14315952,14315958,14316046 3.1131 0.1949 0.0118 10.1016/j.fsigen.2020.102255
```
Comment (RyanBerger98) [inline microhapdb/tests/test_cli.py:340]: `mh18ZHA-004` is a new marker that has been added that is contained within the region of this test. It has been updated to include the new marker.

File: microhapdb/tests/test_cli.py (hunk 1 above)
Comment (standage) [reply microhapdb/tests/test_cli.py:112]: Lol, glad you were able to decipher this.

File: microhapdb/tests/test_cli.py [hunk 5]
```diff
@@ -361,6 +363,7 @@ def test_hg37(capsys):
 mh18PK-87558 MHDBM-1e5374f1    GRCh37 chr18 1960543,1960558,1960562,1960567,1960583,1960589 2.3659 0.1325 0.0494 10.1016/j.fsigen.2018.05.008
 mh18USC-18pA MHDBM-56dfa93b    GRCh37 chr18         5280016,5280019,5280069,5280070,5280092 3.4330 0.2130 0.1818 10.1016/j.fsigen.2019.102213
   mh18CP-005 MHDBM-a85754d3    GRCh37 chr18                 8892862,8892891,8892894,8892905 3.6722 0.0904 0.0059                       ALFRED
+ mh18ZHA-004 MHDBM-db649dba    GRCh37 chr18             14315930,14315951,14315957,14316045 2.9579 0.1949 0.0118 10.1016/j.fsigen.2020.102255
```
Comment (RyanBerger98) [inline microhapdb/tests/test_cli.py:366]: `mh18ZHA-004` is a new marker that has been added that is contained within the query of this test. It has been updated to include the new marker.

File: microhapdb/tests/test_cli.py (hunk 3 above)
Comment (standage) [reply microhapdb/tests/test_cli.py:292]: Yeah, for tests like this, it's more of a consistency check than a correctness check. I haven't put much work into carefully accounting for every record in the frequency table.

File: microhapdb/tests/test_frequency.py [hunk 6]
```diff
@@ -14,7 +14,7 @@
 
 
 def test_assumptions():
-    assert len(microhapdb.frequencies) == 54347 + 366 + 137 + 33 + 103 + 66565 + 4737
+    assert len(microhapdb.frequencies) == 54347 + 366 + 137 + 33 + 103 + 66565 + 4737 + 4131
```
Comment (RyanBerger98) [inline microhapdb/tests/test_frequency.py:17]: I got `4131` by doing `total_len(new_frequencies) - total_len(old_frequencies) = (130,420 - 1) - (126,289  - 1)= 4131`

File: microhapdb/tests/test_frequency.py [hunk 7]
```diff
@@ -25,38 +25,38 @@ def test_allele_frequencies():
...
+    96625  mh15CP-003        KHV  A,A,C      0.268
+    96629  mh15CP-003        LWK  A,A,C      0.056
+    96632  mh15CP-003        MSL  A,A,C      0.047
+    96635  mh15CP-003        MXL  A,A,C      0.328
+    96639  mh15CP-003        PEL  A,A,C      0.235
+    96643  mh15CP-003        PJL  A,A,C      0.344
+    96646  mh15CP-003        PUR  A,A,C      0.221
+    96650  mh15CP-003        STU  A,A,C      0.157
+    96654  mh15CP-003        TSI  A,A,C      0.196
+    96657  mh15CP-003        YRI  A,A,C      0.028
```
Comment (RyanBerger98) [inline microhapdb/tests/test_frequency.py:53]: Updated indexes here

File: microhapdb/tests/test_frequency.py [hunk 8]
```diff
@@ -25,38 +25,38 @@ def test_allele_frequencies():
...
     >>> f.query('Marker == "mh15CP-003" and Allele == "A,A,C" and Population == "FIN"')
                Marker Population Allele  Frequency
-    93553  mh15CP-003        FIN  A,A,C      0.318
+    96602  mh15CP-003        FIN  A,A,C      0.318
```
Comment (RyanBerger98) [inline microhapdb/tests/test_frequency.py:56]: Updated index

File: microhapdb/tests/test_cli.py (hunk 3 above)
Comment (standage) [reply microhapdb/tests/test_cli.py:292]: Although, 130420 - 126289 = 4131, which is reassuring to see below.

File: microhapdb/tests/test_frequency.py [hunk 9]
```diff
@@ -25,38 +25,38 @@ def test_allele_frequencies():
...
     """
     af = microhapdb.frequencies
-    assert af.shape == (126288, 4)
+    assert af.shape == (130419, 4)
```
Comment (RyanBerger98) [inline microhapdb/tests/test_frequency.py:59]: `total_len(frequencies) - 1`

File: microhapdb/tests/test_marker.py [hunk 10]
```diff
@@ -25,7 +25,7 @@ def test_standardize_ids():
 
 
 def test_assumptions():
-    assert len(microhapdb.markers) == 198 + 15 + 40 + 26 + (11 - 1) + 10 + 118 + 90
+    assert len(microhapdb.markers) == 198 + 15 + 40 + 26 + (11 - 1) + 10 + 118 + 90 + 25
```
Comment (RyanBerger98) [inline microhapdb/tests/test_marker.py:28]: The addition of the 25 new markers

File: microhapdb/tests/test_marker.py [hunk 11]
```diff
@@ -34,24 +34,24 @@ def test_markers():
...
-    30  mh01KK-117  MHDBM-39dc025f    GRCh38  chr1  204664211,204664268,204664371,204664397  4.4565  0.1933  0.0472  ALFRED
+    31  mh01KK-117  MHDBM-39dc025f    GRCh38  chr1  204664211,204664268,204664371,204664397  4.4565  0.1933  0.0472  ALFRED
     >>> m[m.Name == 'mh11PK-63643']
                  Name          PermID Reference  Chrom                                            Offsets  Ae  In  Fst                        Source
-    272  mh11PK-63643  MHDBM-c5ce121f    GRCh38  chr11  34415814,34415816,34415818,34415835,34415836,3... NaN NaN  NaN  10.1016/j.fsigen.2018.05.008
+    287  mh11PK-63643  MHDBM-c5ce121f    GRCh38  chr11  34415814,34415816,34415818,34415835,34415836,3... NaN NaN  NaN  10.1016/j.fsigen.2018.05.008
     >>> m[m.Name == 'mh02AT-05']
              Name          PermID Reference Chrom                        Offsets      Ae   In    Fst         Source
-    67  mh02AT-05  MHDBM-c3feaba8    GRCh38  chr2  160222899,160222923,160222938  4.5544  0.2  0.152  ISFG2019:P597
+    73  mh02AT-05  MHDBM-c3feaba8    GRCh38  chr2  160222899,160222923,160222938  4.5544  0.2  0.152  ISFG2019:P597
```
Comment (RyanBerger98) [inline microhapdb/tests/test_marker.py:46]: Updated index

File: microhapdb/tests/test_marker.py [hunk 12]
```diff
@@ -34,24 +34,24 @@ def test_markers():
...
     """
     m = microhapdb.markers
     vm = microhapdb.variantmap
-    assert m.shape == (507, 9)
+    assert m.shape == (532, 9)
```
Comment (RyanBerger98) [inline microhapdb/tests/test_marker.py:50]: The length has increased by 25 with the addition of the new markers

File: microhapdb/tests/test_marker.py [hunk 13]
```diff
@@ -34,24 +34,24 @@ def test_markers():
...
     result = m[m.Chrom == 'chr19']
-    assert len(result) == 13
+    assert len(result) == 14
```
Comment (RyanBerger98) [inline microhapdb/tests/test_marker.py:52]: One more marker was added that is located on chromosome 19

File: microhapdb/tests/test_marker.py [hunk 14]
```diff
@@ -491,7 +491,7 @@ def test_amplicon_object(capsys):
     ('mh04KK-010', 'mh04KK-010 MHDBM-07c8d144    GRCh38  chr4 1985210,1985244 2.9226 0.1306 0.0406 ALFRED'),
     ('mh08PK-46625', 'mh08PK-46625 MHDBM-840756f3    GRCh38  chr8 1194352,1194356,1194364,1194371 2.3775 0.1132 0.1395 10.1016/j.fsigen.2018.05.008'),
     ('mh04AT-10', 'mh04AT-10 MHDBM-07c8d144    GRCh38  chr4 1985210,1985244 2.9226 0.1306 0.0406 ISFG2019:P597'),
-    ('mh01NH-03', 'mh01NH-03 MHDBM-e7a95c5e    GRCh38  chr1 184807944,184807966,184808042 1.9086 0.1987 0.1725 10.1016/j.legalmed.2015.06.003'),
+    ('mh01NH-03', 'mh01NH-03 MHDBM-e7a95c5e    GRCh38  chr1 184807944,184807966,184808042 1.9085 0.1987 0.1725 10.1016/j.legalmed.2015.06.003'),
```
Comment (RyanBerger98) [inline microhapdb/tests/test_marker.py:494]: Rounding difference

**Review by standage — APPROVED** [review]
Looks great. I'm gonna do a quick update to the README and then merge.

**Pull request merged**